aipython/stripsCSPPlanner.py

```diff
@@ -59,9 +77,73 @@ def __init__(self, planning_problem, horizon=2):
+                              for (var, val) in initial_state.items()})
+            #goal state positions
+            positions.update({Constraint((st(var, horizon),),
+                                         is_(val)).__repr__(): (maxcurrwidth + 50, old_pos['action'][1])
+                              for (var, val) in goal.items()})
+            #precondition positions
+            positions.update({Constraint((st(var, stage), st('action', stage)),
+                                         if_(val, act)).__repr__(): (minwidth + (diff) * (2 * (stage + 1) - 1.5), old_pos['action'][1])
+                              for act, strps in prob_domain.strips_map.items()
+                              for var, val in strps.preconditions.items()
+                              for stage in range(horizon)})        
+            #effect positions
+            positions.update({Constraint((st(var, stage + 1), st('action', stage)),
+                                         if_(val, act)).__repr__(): (minwidth + (diff) * (2 * (stage + 1) - 0.5), old_pos['action'][1])
+                              for act, strps in prob_domain.strips_map.items()
+                              for var, val in strps.effects.items()
+                              for stage in range(horizon)})
+
+            #frame positions
+            positions.update({Constraint((st(var, stage), st('action', stage), st(var, stage + 1)),
+                                         eq_if_not_in_({act for act in prob_domain.actions
+                                                        if var in prob_domain.strips_map[act].effects})).__repr__(): (old_pos[var][0] + (diff) * (2 * (stage + 1) - 0.75), old_pos[var][1])
+                              for var in prob_domain.feats_vals
+                              for stage in range(horizon)})
+
+
+            pos_keys = list(positions.keys())
+            for i in range(len(pos_keys)):
+                for j in range(i + 1, len(pos_keys)):
+                    if positions[pos_keys[i]] == positions[pos_keys[j]]:
+                        positions[pos_keys[j]] = (
+                            positions[pos_keys[j]][0], positions[pos_keys[j]][1] - 200)
+        except:
+                pass
+        # Numbering the conditions with the same names: and order them vertically    
+        constraints = sorted(constraints, key=lambda con: con.__repr__())
+        constraint_strings = list(
+            map(lambda con: con.__repr__(), constraints))
+        i = 0
+        while i < len(constraint_strings):
+            occurence = constraint_strings.count(constraint_strings[i])
+            if occurence > 1:
+                old_repr = constraints[i].repr
+                if old_repr in positions:
+                    old_posx, old_posy = positions[old_repr]
+                    
+                for j in range(occurence):
+                    # start numbering
+                    gap = min(200 / occurence, 50)
```

Comment:
please write a comment here as it is hard to understand what is `gap` for.

Reply:
I have Changed the name of the constraint. Therefore, it is no longer needed to handle the duplicated names, and I have removed these code.